File: 1-Two-Sum/solution.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        rslt = {}

        for i, num in enumerate(nums):
            if num in rslt:
                return (i, rslt[num])
            else:
                rslt[target - num] = i


foo = Solution()
print(foo.twoSum([2, 7, 11, 15], 9))

File: 1-Two-Sum/solution.go
func twoSum(nums []int, target int) []int {
	rslt := []int{0, 0}
	for idx_1, v_1 := range nums {
		for idx_2, v_2 := range nums[idx_1+1:] {
			if v_1+v_2 == target {
				rslt[0] = idx_1
				rslt[1] = idx_2 + idx_1 + 1
				break
			}
		}
	}
	return rslt
}
